include/dsn/tool-api/logging_provider.h

```diff
@@ -87,6 +86,8 @@ class logging_provider
     static logging_provider *create_default_instance();
 };
 
-/*@}*/
-// ----------------------- inline implementation ---------------------------------------
+namespace tools {
+void set_log_prefixed_message_func(std::function<std::string()> func);
+} // namespace tools
+
```

Comment:
```suggestion
void set_log_prefixed_message_func(std::function<std::string()> func);
```

No need to keep the meaningless `namespace tools`.